logseq/logseq
# Feature: Add named block links for web and publishing

Per https://linear.app/logseq/issue/LOG-2047/add-permalink-to-blocks-in-published-graph, this PR adds permanent named block links to header blocks for web and publishing apps e.g.
<img width="591" alt="Screen Shot 2022-12-09 at 5 27 17 PM" src="https://user-images.githubusercontent.com/97210743/206805873-933ac405-06cc-488b-81f4-25c09a625d2f.png">

To get the url of a heading block, simply zoom into the block and get it from the address bar. The block name that is derived from the heading is case insensitive, which allows for changing a header's case without effecting the resulting url. A named block link looks like https://docs.logseq.com/#/page/Logseq%20Sync/block/faq. Since these links have page context, the heading block is only unique per page

To test this:
* Build publishing asset - `clojure -M:cljs release publishing`
* Make sure graph has some heading blocks
* Run command `Export public graph...`
* open published index.html
* Zoom in on heading blocks and observe urls change as described
* Open block url in new tab and it opens correctly

**Question**:
Some feedback I'm seeking:
Would it be more useful if
a) any headings on a page are automatically linkable like github
OR
b) only certain blocks are linkable when set with an explicit property?

Currently the feature works as b) but it wouldn't be hard to switch to a). For an example of a, the three headers in https://docs.logseq.com/#/page/Logseq%20Sync, Usage, Functionality and FAQ would have a link like https://docs.logseq.com/#/page/Logseq%20Sync/block/faq

**Answer**:
Went with a) after some discussion with team
